packages/swap/src/routes/quote.ts

```diff
@@ -84,167 +101,147 @@ const quote = (io: Server) => {
 
       if (query.boostFeeBps) {
         const boostFee = getPipAmountFromAmount(swapInputAmount, query.boostFeeBps);
-        includedFees.push({
-          type: 'BOOST',
-          chain: srcChainAsset.chain,
-          asset: srcChainAsset.asset,
-          amount: boostFee.toString(),
-        });
+        includedFees.push(buildFee(srcAsset, 'BOOST', boostFee));
         swapInputAmount -= boostFee;
       }
 
-      let ingressFee = await getIngressFee(srcChainAsset);
+      let ingressFee = await getIngressFee(srcAsset);
       if (ingressFee == null) {
-        throw ServiceError.internalError(
-          `could not determine ingress fee for ${getInternalAsset(srcChainAsset)}`,
-        );
+        throw ServiceError.internalError(`could not determine ingress fee for ${srcAsset}`);
       }
       if (ingressEgressFeeIsGasAssetAmount) {
-        ingressFee = await estimateIngressEgressFeeAssetAmount(
-          ingressFee,
-          getInternalAsset(srcChainAsset),
-        );
+        ingressFee = await estimateIngressEgressFeeAssetAmount(ingressFee, srcAsset);
       }
-      includedFees.push({
-        type: 'INGRESS',
-        chain: srcChainAsset.chain,
-        asset: srcChainAsset.asset,
-        amount: ingressFee.toString(),
-      });
+      includedFees.push(buildFee(srcAsset, 'INGRESS', ingressFee));
       swapInputAmount -= ingressFee;
       if (swapInputAmount <= 0n) {
         throw ServiceError.badRequest(`amount is lower than estimated ingress fee (${ingressFee})`);
       }
 
       if (query.brokerCommissionBps) {
         const brokerFee = getPipAmountFromAmount(swapInputAmount, query.brokerCommissionBps);
-        includedFees.push({
-          type: 'BROKER',
-          chain: srcChainAsset.chain,
-          asset: srcChainAsset.asset,
-          amount: brokerFee.toString(),
-        });
+        includedFees.push(buildFee(srcAsset, 'BROKER', brokerFee));
         swapInputAmount -= brokerFee;
       }
 
-      const quotePools = await getPools(
-        getInternalAsset(srcChainAsset),
-        getInternalAsset(destChainAsset),
+      const poolQuote = getPoolQuoteResult(
+        srcAsset,
+        destAsset,
+        swapInputAmount,
+        ingressEgressFeeIsGasAssetAmount,
+        includedFees,
+      );
+
+      const pools = await getPools(srcAsset, destAsset);
+
+      const firstLegPoolFee = getHundredthPipAmountFromAmount(
+        swapInputAmount,
+        pools[0].liquidityFeeHundredthPips,
       );
 
-      const quoteRequest = buildQuoteRequest({
-        ...query,
-        amount: String(swapInputAmount),
-      });
-      io.emit('quote_request', quoteRequest);
+      includedFees.push(buildFee(srcAsset, 'LIQUIDITY', firstLegPoolFee));
+
+      swapInputAmount -= firstLegPoolFee;
```

Comment:
was a bit confused about this, maybe we can move it down after the `if` (i think its not relevant for that block, right?)